analyses/cell-type-neuroblastoma-04/scripts/setup/prepare-gene-lists.R

```diff
@@ -43,74 +49,95 @@ option_list <- list(
   dplyr::select(gene_ids, gene_symbol_scpca)
 
-# Map over sheets to get genes, keeping only those we have in ScPCA
-nbatlas_markers_df <- sheet_names |>
+
+# Map over relevant sheets in both spreadsheet to get genes
+
+
+# First, define a helper function to prepare an excel sheet for joining
+# sheet_name: The name of the sheet to read in
+# excel_file_path: Path to excel file with the given sheet
+prepare_excel_sheet <- function(sheet_name, excel_file_path) {
+  readxl::read_excel(excel_file_path, sheet = sheet_name) |>
+    # keep only genes we have
+    dplyr::inner_join(scpca_gene_table, by = c("gene" = "gene_symbol_scpca")) |>
+    # keep pval and LFC so we can rank/subset later as needed
+    dplyr::select(ensembl_gene_id = gene_ids, gene_symbol = gene, p_val_adj, avg_log2FC)
+}
+
+
+s2_df <- s2_sheet_names |>
+  purrr::map(
+    prepare_excel_sheet,
+    opts$table_s2_excel_file
+  ) |>
+  purrr::list_rbind(names_to = "NBAtlas_label")
+
+s5_df <- s5_sheet_names |>
   purrr::map(
-    \(x) {
-      readxl::read_excel(opts$excel_file, sheet = x) |>
-        # keep only genes we have
-        dplyr::inner_join(scpca_gene_table, by = c("gene" = "gene_symbol_scpca")) |>
-        # keeping pval and LFC will allow us to rank/subset later as needed
-        dplyr::select(ensembl_gene_id = gene_ids, gene_symbol = gene, p_val_adj, avg_log2FC)
-    }
+    prepare_excel_sheet,
+    opts$table_s5_excel_file
   ) |>
-  purrr::list_rbind(names_to = "NBAtlas_label") |>
+  purrr::list_rbind(names_to = "NBAtlas_label")
+
+# Combine data frames and finalize processing
+
+
+nbatlas_markers_df <- s2_df |>
+  dplyr::bind_rows(s5_df) |>
   dplyr::mutate(
     # indicate if up or down-regulated, since there are positive and negative LFCs
     direction = ifelse(avg_log2FC > 0, "up", "down"),
     source = "NBAtlas",
-    # recode NE -> Neuroendocrine
-    NBAtlas_label = ifelse(NBAtlas_label == "NE", "Neuroendocrine", NBAtlas_label)
+    # recode some labels to match what is in the objects
+    NBAtlas_label = dplyr::case_when(
+      NBAtlas_label == "NE" ~ "Neuroendocrine",
+      NBAtlas_label == "Cycling" ~ "Immune cycling",
+      NBAtlas_label == "TOX-KIT NK cell" ~ "TOX2+/KIT+ NK cell",
+      .default = NBAtlas_label)
   )
 
 # export
 readr::write_tsv(nbatlas_markers_df, opts$nbatlas_marker_gene_file)
 
 #### cell-type-consensus marker genes --------------------------
 
-# map of names to match NBAtlas labels with our consensus labels (excluding NE)
-# consensus = nbatlas
-celltype_map <- c(
-  "B cell" = "B cell",
-  "endothelial cell" = "Endothelial",
-  "fibroblast" = "Fibroblast",
-  "myeloid" = "Myeloid",
-  "natural killer cell" = "NK cell",
-  "dendritic cell" = "pDC",
-  "plasma cell" = "Plasma",
-  "stromal cell" = "Stromal other", # may not be a great match, but can't hurt
-  "T cell" = "T cell"
-)
-
 validation_df <- readr::read_tsv(opts$consensus_marker_genes_url)
 
 consensus_markers_df <- validation_df |>
```

Comment:
Is there a reason we need to even have this file here at all if we aren't using the NBatlas label? Can't we just read it in using the URL in the `cell-type-consensus` module when we need it?

Reply:
Yeah, this makes more sense to just grab when needed since it's so small.